rfc/rfc-0009-supply-chain-switches-template-on-flag.md

```diff
@@ -1,3 +1,182 @@
+to choose between templates within a step.
+
+## Motivation
+
+The Developer Productivity program auto-generates workloads from CLI commands.
+Those workloads match with supplychains and templates that are written by the
+dev-prod team as well. The team would like to limit the number of supplychains
+that must be written and maintained. Small differences in workload can lead to
+needing an entirely different supply chain. For example, what code source is used.
+
+Further info on their use case can be read [here](https://docs.google.com/document/d/1TVqlNqTyCMlp_yNs9_F80QAIQSC5bxm05vSyWyeGMSw/edit).
+See also this [discussion](https://vmware.slack.com/archives/C01UX69LJCB/p1629408443051200).
+
+## Explanation
+
+### Current state of Cartographer
+
+Cartographer is already capable of matching a supply chain to a workload through selectors.
+With no further effort, a platform can be created that has multiple supply chains with different
+purposes or requirements. By specifying characteristics of workloads in the workload labels,
+devs can provide information to operators necessary to choose the right supply chain for
+each workload.
+
+While this behavior is currently possible, relying only on this selection would lead to a platform
+with a large number of supply chains.
+
+Workarounds, for example templates that use ytt templating to embed conditional behavior,
+would represent a code smell that this primitive is not adequate for Cartographer users.
+
+### Condition template choice on information available from the workload
+
+When creating objects of the supply chain, Cartographer has the workload definition available.
+Allowing the supply chain to leverage the workload would add additional power to the conditional.
+
+```yaml
+apiVersion: kontinue.io/v1alpha1
+kind: ClusterSupplyChain
+metadata:
+  name: responsible-ops
+spec:
+  selector:
+    app: web
+  components:
+    - name: source-provider
+      templateRef:
+        kind: ClusterSourceTemplate
+        options:                                  # <--- a list
+        - name: git-template
+          matchExpressions:
+            - key: workload.spec.source.git       # <--- path to field in template context
+              operator: Exists
+        - name: imgpkg-bundle-template
+          matchExpressions:
+            - key: workload.spec.source.image       # <--- path to field in template context
+              operator: Exists
+
+---
+apiVersion: kontinue.io/v1alpha1
+kind: Workload
+metadata:
+  name: petclinic
+  labels:
+    app: web
+spec:
+  source:
+    git:                                        # <--- Presence of this field will determine the conditional
+      url: https://github.com/spring-projects/spring-petclinic.git
+      ref:
+        branch: main
+```
+
+1. As labels are present in the workload metadata, any behavior possible using just label selectors
+  will be possible using the entire templating context.
+
+## Open choices
+
+1. Should options be an ordered list?
```

Comment:
Personally, I think a (helpful) error would be least surprising for users. If we do want an ordered list, though, I'd suggest we need some way of communicating that better in the spec, as I don't know that `options` conveys an ordered list.